### PR TITLE
:sparkles: (dashboards) ability to save net-worth widget details

### PR DESCRIPTION
**Pull request by MatissJanis**
Added the ability to save filters & date-ranges on the net worth widgets.

How to use:
1. open reports page
2. open a net worth graph (click on it)
3. in the net worth page: change date-range and maybe add a filter
4. click save
5. go back to dashboard
6. the widget will show the filtered data and with the specific time-range

Feedback is welcome. I plan to implement the same behaviour to the other existing reports too, but starting with just this one.

https://github.com/actualbudget/actual/issues/3282

### CODE REVIEW

File: upcoming-release-notes/3364.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+category: Enhancements
+authors: [psybers]
```
Comment (MikesGlitch) [inline upcoming-release-notes/3364.md:3]: Identity theft 😄

**Comment by MikesGlitch** [comment]
I managed to break it by changing the filters from the default "April 2024 - September 2024" to 

sfdafda

File: upcoming-release-notes/3364.md (hunk 1 above)
Comment (MatissJanis) [reply upcoming-release-notes/3364.md]: oops :D

**Comment by MatissJanis** [comment]
@youngcw - great catch. I forgot to add filters to the dashboard cards 🤦 

@MikesGlitch - this looks like a pre-existing issue (visible on edge too). I'll open a separate issue for patching this.

**Review by MikesGlitch — DISMISSED** [review]

**Comment by youngcw** [comment]
Will the date ranges update automatically still?

**Comment by MatissJanis** [comment]
> Will the date ranges update automatically still?

Depends how you select them. If you select a specific date-range via the dropdowns: that will remain static. However, if you select via the buttons (i.e. "past year") - that will auto-magically update.

Now that I'm tying this.. it would make sense to display somewhere an indicator if the selected timeframe is static or a sliding window. Any UX suggestions for that?

**Comment by youngcw** [comment]
You could always make it like the custom reports where you have to decide if its fixed or dynamic.

**Comment by carkom** [comment]
> > Will the date ranges update automatically still?
> 
> Depends how you select them. If you select a specific date-range via the dropdowns: that will remain static. However, if you select via the buttons (i.e. "past year") - that will auto-magically update.
> 
> Now that I'm tying this.. it would make sense to display somewhere an indicator if the selected timeframe is static or a sliding window. Any UX suggestions for that?

Something like this?

https://github.com/carkom/actual/blob/bf729ca17bd83434f776de70ddb96e3541699968/packages/desktop-client/src/components/reports/Header.jsx

**Comment by MatissJanis** [comment]
Good idea @carkom ! I used your idea and then slightly built upon that by introducing a play/pause button. That allows us to save vertical space by having just a single button instead of two (+ margins between them).

A question you folks might now ask: why do we still show both the time-frame selector (dropdowns), but also the quick-pick buttons?

I think it makes sense to keep both of these visible at all times. This allows the user to also select, for example, "last 7 months" and then make it "live update". It also allows to quickly select "last 3 months" and then "pause" the timeFrame. Quite a handy feature if you ask me :) 

Also: the quick-pick and static selection will automatically update the play/pause state. You can always swap it back.. but by default IMO the quick-pick options should have "play" enabled. And same goes for static selections - those should have "paused" by default, but allow to switch it to "play" if the user wishes so.


https://github.com/user-attachments/assets/4f94bc5b-26dd-4620-9cfa-492393ae4515

**Review by MikesGlitch — APPROVED** [review]

**Comment by MatissJanis** [comment]
Updated `Header` component to TS while I'm working in it here.. out for review again!

**Comment by youngcw** [comment]
> I think it makes sense to keep both of these visible at all times. This allows the user to also select, for example, "last 7 months" and then make it "live update". It also allows to quickly select "last 3 months" and then "pause" the timeFrame. Quite a handy feature if you ask me :)

Maybe we could look into compressing the quick selections down into a menu or something in the future, but for now it looks good.

**Comment by carkom** [comment]
Looks pretty good! 

@MatissJanis can you move filter button to the far right? I'm not sure why it bothers me where it is but it does... 😬

**Comment by MatissJanis** [comment]
> Maybe we could look into compressing the quick selections down into a menu or something in the future, but for now it looks good.

@youngcw that's a good idea for future when we need more vertical space there. But for now I think we're good :)

--



> @MatissJanis can you move filter button to the far right? I'm not sure why it bothers me where it is but it does... 😬

Good idea! Moved them to the far right

**Review by MikesGlitch — APPROVED** [review]

**Review by youngcw — APPROVED** [review]

**Pull request merged**